src/preferences/node/preference-server-module.ts

```diff
@@ -8,28 +8,50 @@
 import { ContainerModule, } from 'inversify';
 import { ConnectionHandler, JsonRpcConnectionHandler } from '../../messaging/common';
 import { IPreferenceServer, IPreferenceClient } from '../common/preference-protocol'
+import { WorkspaceServer } from '../../workspace/common';
 import { DefaultPreferenceServer } from '../node/default-preference-server'
 import { CompoundPreferenceServer } from '../common/compound-preference-server'
-import { JsonPreferenceServer, PreferencePath } from './json-preference-server'
+import { JsonPreferenceServer, WorkspacePreferenceServer, UserPreferenceServer, PreferencePath } from './json-preference-server'
 import { FileUri } from "../../application/node/file-uri";
 
 import * as path from 'path';
 
 export const preferenceServerModule = new ContainerModule(bind => {
-    const WorkspacePreferenceServer = Symbol('WorkspacePreferenceServer');
-    const UserPreferenceServer = Symbol('UserPreferenceServer');
-
     bind(DefaultPreferenceServer).toSelf().inSingletonScope();
+    bind(JsonPreferenceServer).toSelf();
+
+    // Workspace preference server that watches the current workspace
+    bind(WorkspacePreferenceServer).toDynamicValue(ctx => {
+        const workspaceServer = ctx.container.get<WorkspaceServer>(WorkspaceServer);
+        const preferencePath = workspaceServer.getRoot().then(root => {
+            path.resolve(root, '.theia', 'prefs.json');
```

Comment:
`FileUri.create(root).resolve('.theia', 'prefs.json')`, since an expected type is `URI` not `string`